File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnSurfaceVariantDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSurfaceVariantDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd6
            AppTheme.STANDARD -> R.color.cstd6
            AppTheme.BLUE -> R.color.cbd6
            AppTheme.GREEN -> R.color.cgd6
            AppTheme.RED -> R.color.crd6
            AppTheme.NOTHING -> R.color.cnod6
        }
    }
}

File: lib/layer2_core/dark/e_on_surface_variant_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnSurfaceVariantDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd6;
      case 'standard': return cstd6;
      case 'blue': return cbd6;
      case 'green': return cgd6;
      case 'red': return crd6;
      case 'nothing': return cnod6;
      default: return cstd6;
    }
  }
}
